k8sutils/services.go

```diff
@@ -172,16 +176,27 @@ func CreateOrUpdateService(namespace string, serviceMeta metav1.ObjectMeta, owne
 // patchService will patch Redis Kubernetes service
 func patchService(storedService *corev1.Service, newService *corev1.Service, namespace string) error {
 	logger := serviceLogger(namespace, storedService.Name)
-	patchResult, err := patch.DefaultPatchMaker.Calculate(storedService, newService, patch.IgnoreStatusFields())
+	// We want to try and keep this atomic as possible.
+	newService.ResourceVersion = storedService.ResourceVersion
+	newService.CreationTimestamp = storedService.CreationTimestamp
+	newService.ManagedFields = storedService.ManagedFields
+
+	if newService.Spec.Type == generateServiceType("ClusterIP") {
+		newService.Spec.ClusterIP = storedService.Spec.ClusterIP
+	}
+
+	patchResult, err := patch.DefaultPatchMaker.Calculate(storedService, newService,
+		patch.IgnoreStatusFields(),
+		patch.IgnoreField("kind"),
+		patch.IgnoreField("apiVersion"),
```

Comment:
Without these, I was always getting a change, when they were identically set. This removes it from being considered within the patch.